src/Microsoft.Azure.WebJobs.Host.Storage/StorageServiceCollectionExtensions.cs

```diff
@@ -24,15 +24,18 @@ public static void AddAzureStorageCoreServices(this IServiceCollection services)
             // Add runtime services that depend on storage.
             services.AddSingleton<IDistributedLockManager>(provider => Create(provider));
 
-            services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<JobHostInternalStorageOptions>, CoreWebJobsOptionsSetup<JobHostInternalStorageOptions>>());
-
             services.TryAddSingleton<IDelegatingHandlerProvider, DefaultDelegatingHandlerProvider>();
 
             // Adds necessary Azure services to create clients
             services.AddAzureClientsCore();
 
-            services.TryAddSingleton<IAzureBlobStorageProvider, AzureStorageProvider>();
+            services.AddAzureStorageScaleServices();
```

Comment:
I think it'd be safer to just duplicate the registrations in the new method rather than call the new API here. In the future we might add unrelated services to the new builder method that we DONT want to include in core scenarios. This configuration makes it easy to make a mistake.